src/go/MONIT/monit.go

```diff
@@ -576,6 +579,17 @@ func parseStats(data map[string]interface{}, verbose int) []Record {
 	return out
 }
 
+// helper function to deal with stats query
+func getStats(short_term_url string, long_term_url string, t string, short_term_dbid int, long_term_dbid int, database string, q string, esapi string, idx int, limit int, verbose int, stompConfig StompConfig, inject bool, hdfs string) {
```

Comment:
You should define a struct with all of these parameters and pass it around, e.g.
```
type StatsParams struct {
   ShortUrl string
   LongUrl string
}
func getStats(params StatsParams) {
    ...
    long_term_data := run(params.LongURL, ...)
}
```
Doing this way your code will be much easier to read, and you can have as many parameters to pass into function as you like, just extend the struct.